src/server.js

```diff
@@ -95,7 +151,73 @@ export class Server {
       `tx=${tx}`,
       { timeout: SUBMIT_TRANSACTION_TIMEOUT }
     )
-      .then((response) => response.data)
+      .then((response) => {
+        if (!response.data.result_xdr) {
+          return response.data;
+        }
+
+        const responseXDR = xdr.TransactionResult.fromXDR(
+          response.data.result_xdr,
+          'base64'
+        );
+        const results = responseXDR.result().value();
+
+        let offerResults;
+        let hasManageOffer;
+
+        if (results.length) {
+          offerResults = results
+            .map((result, i) => {
+              if (typeof result.value().value().success !== 'function') {
+                return null;
+              }
+              hasManageOffer = true;
+
+              // yeah this is annoying
+              const offerSuccess = result
+                .value()
+                .value()
+                .success();
+
+              const offersClaimed = offerSuccess
+                .offersClaimed()
+                .map((offerClaimed) => ({
+                  sellerId: offerClaimed
+                    .sellerId()
+                    .value()
+                    .toString('hex'),
+                  offerId: offerClaimed.offerId,
+                  assetSold: offerClaimed.assetSold().switch(),
+                  amountSold: offerClaimed.amountSold,
+                  assetBought: offerClaimed.assetSold().switch(),
```

Comment:
This looks like it might be a copy-paste error here. Should it say something like `assetBought` on this line and `amountBought` on the next line?